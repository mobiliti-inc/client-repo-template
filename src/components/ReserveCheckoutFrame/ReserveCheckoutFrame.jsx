/* eslint-disable camelcase */// API returns lots of snake_case data

import PropTypes from 'prop-types';

import { backArrow } from '../../assets/icons';
import styles from './ReserveCheckoutFrame.scss';
import Tile from '../Tile/Tile';
import VehicleTitle from '../VehicleTitle/VehicleTitle';
import VehicleCostBreakdown from '../../pages/Vehicles/Detail/VehicleCostBreakdown/VehicleCostBreakdown';
import VehicleDealer from '../../pages/Vehicles/Detail/VehicleDealer/VehicleDealer';
import VehicleModel from '../../models/Vehicle';

class ReserveCheckoutFrame extends React.Component {
	static propTypes = {
		vehicle: VehicleModel.isRequired,
		pageTitle: PropTypes.string.isRequired,
		history: PropTypes.shape({
			goBack: PropTypes.func
		}).isRequired,
		location: PropTypes.shape({
			data: PropTypes.shape({
				latitude: PropTypes.number,
				longitude: PropTypes.number
			}),
			geolocationUnavailable: PropTypes.bool
		}).isRequired,
		getUserLocation: PropTypes.func.isRequired,
		children: PropTypes.node.isRequired
	}
	componentWillMount() {
		window.scrollTo({ top: 0, behavior: 'smooth' });
	}

	render() {
		const {
			vehicle_title,
			objImages,
			year_of_make,
			vehicle_model,
			vehicle_make,
			vehicle_status,
			subscription_price,
		} = this.props.vehicle;
		const costBreakdown = [
			{ name: '', price: subscription_price },
		]; // @TODO: get real cost breakdown
		const imageUrl = objImages[0].image_url;
		return (
			<div styleName="container">
				<div styleName="left">
					<div styleName="reserve-nav">
						<button
							onClick={this.props.history.goBack}
							styleName="back-arrow"
						>
							<img src={backArrow} alt="Back" />
						</button>
						<p>{this.props.pageTitle}</p>
					</div>
					<Tile className="lighter-tile">
						<img styleName="vehicle-image" src={imageUrl} alt={`${vehicle_title}`} />
						<div styleName="vehicle-details-container">
							<div>
								<div styleName="vehicle-tile-container">
									<VehicleTitle
										yearOfMake={year_of_make}
										vehicleModel={vehicle_model}
										vehicleMake={vehicle_make}
										vehicleStatus={vehicle_status}
									/>
								</div>
								<VehicleCostBreakdown
									data={costBreakdown}
									removeBorderBottom
								/>
							</div>
						</div>
					</Tile>

					<VehicleDealer
						vehicle={this.props.vehicle}
						location={this.props.location}
						getUserLocation={this.props.getUserLocation}
					/>
				</div>
				<div styleName="right">
					{this.props.children}
				</div>
			</div>
		);
	}
}

export default CSSModules(ReserveCheckoutFrame, styles);
export { ReserveCheckoutFrame };
