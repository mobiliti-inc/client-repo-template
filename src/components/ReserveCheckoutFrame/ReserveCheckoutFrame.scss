@import '~variables';

.container {
  display: flex;
  flex-direction: column;
  max-width: 1264px;
  width: 100%;
  height: 100%;
  position: relative;

  @media (min-width: $tablet-width) {
    padding-left: 30px;
    display: grid;
    grid-template-columns: 35% 65%;
  }
}

.left {
  position: relative;
  @media (min-width: $tablet-width) {
		right: 0;
  }
  .reserve-nav {
    display: flex;
    align-items: center;
    padding-left: 15px;
    @media (min-width: $tablet-width) {
      padding-left: 0;
    }
    p {
      color: #0A1426;
      font-size: 22px;
      font-family: $futura;
      font-weight: bold;
      padding-left: 20px;
    }
    .back-arrow {
      margin: 30px 0;
      cursor: pointer;
      border: none;
      padding: 0;
      background: none;
      img {
        transition: all .2s linear;
        box-shadow: 1px 0 8px rgba(228, 228, 228, 0.514);
      }
      &:focus {
        outline: none;
      }
      &:hover {
        img {
          box-shadow: none;
        }
      }
    }
  }
  .vehicle-image {
    width: 100%;
  }
  .vehicle-tile-container {
    padding-bottom: 20px;
  }
  .vehicle-details-container {
    padding-bottom: 20px;
    display: flex;
    flex-direction: column;
    align-items: center;
    @media (min-width: $tablet-width) {
      align-items: flex-start;
    }
    @media (max-width: $half-mobile-width) {
      padding-left: 15px;
    }
    
  }
}

.right {
  &:not(div:nth-last-child()) {
    margin-bottom: 10px;
  }
  position: relative;
  padding: 30px 15px 25px;
	@media (min-width: $tablet-width) {
		padding: 105px 30px 0;
	}
	margin-left: 0;
}
