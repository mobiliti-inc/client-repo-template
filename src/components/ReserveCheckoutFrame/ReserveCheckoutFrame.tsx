import { backArrow } from "../../assets";
import styles from "./ReserveCheckoutFrame.scss";
import Tile from "../Tile/Tile";
import VehicleTitle from "../VehicleTitle/VehicleTitle";
import VehicleCostBreakdown from "../../pages/Vehicles/Detail/VehicleCostBreakdown/VehicleCostBreakdown";
import VehicleDealer from "../../pages/Vehicles/Detail/VehicleDealer/VehicleDealer";
import VehicleModel from "../../models/Vehicle";

type ReserveCheckoutFrameProps = {
	vehicle: any,
	pageTitle: string,
	history: {
		goBack?: (...args: any[]) => any
	},
	location: {
		data?: {
			latitude?: number,
			longitude?: number
		},
		geolocationUnavailable?: boolean
	},
	getUserLocation: (...args: any[]) => any
};

class ReserveCheckoutFrame extends React.Component<ReserveCheckoutFrameProps, {}> {
	componentWillMount() {
		window.scrollTo({ top: 0, behavior: "smooth" });
	}
	render() {
		const {
			vehicle_title,
			objImages,
			year_of_make,
			vehicle_model,
			vehicle_make,
			vehicle_status,
			subscription_price
		} = this.props.vehicle;
		const costBreakdown = [{ name: "", price: subscription_price }]; // @TODO: get real cost breakdown
		const imageUrl = objImages[0].image_url;
		return (
			<div styleName="container">
				<div styleName="left">
					<div styleName="reserve-nav">
						<button onClick={this.props.history.goBack} styleName="back-arrow">
							<img src={backArrow} alt="Back" />
						</button>
						<p>{this.props.pageTitle}</p>
					</div>
					<Tile className="lighter-tile">
						<img styleName="vehicle-image" src={imageUrl} alt={`${vehicle_title}`} />
						<div styleName="vehicle-details-container">
							<div>
								<div styleName="vehicle-tile-container">
									<VehicleTitle
										yearOfMake={year_of_make}
										vehicleModel={vehicle_model}
										vehicleMake={vehicle_make}
										vehicleStatus={vehicle_status}
									/>
								</div>
								<VehicleCostBreakdown data={costBreakdown} removeBorderBottom />
							</div>
						</div>
					</Tile>

					<VehicleDealer
						vehicle={this.props.vehicle}
						location={this.props.location}
						getUserLocation={this.props.getUserLocation}
					/>
				</div>
				<div styleName="right">{this.props.children}</div>
			</div>
		);
	}
}

export default CSSModules(ReserveCheckoutFrame, styles);
export { ReserveCheckoutFrame };
