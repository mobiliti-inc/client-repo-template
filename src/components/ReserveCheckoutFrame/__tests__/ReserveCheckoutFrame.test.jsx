import { ReserveCheckoutFrame } from '../ReserveCheckoutFrame';
import vehicles from '../../../actions/sampleData/sampleVehicles';

describe('ReserveFrame', () => {
	const vehicle = vehicles[0];
	const props = {
		vehicle,
		history: {
			goBack: jest.fn()
		},
		location: {
			data: {
				latitude: 2344,
				longitude: -3455
			},
		},
		geolocationUnavailable: false,
		getUserLocation: jest.fn(),
		pageTitle: 'Reserve',
		children: <div>Yes</div>
	};
	it('renders', () => {
		const subject = shallow(<ReserveCheckoutFrame {...props} />);
		matchSnapshot(subject);
	});
});
