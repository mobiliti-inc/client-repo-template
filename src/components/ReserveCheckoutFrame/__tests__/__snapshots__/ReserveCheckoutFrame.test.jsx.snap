// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReserveFrame renders 1`] = `
<div
  styleName="container"
>
  <div
    styleName="left"
  >
    <div
      styleName="reserve-nav"
    >
      <button
        onClick={[MockFunction]}
        styleName="back-arrow"
      >
        <img
          alt="Back"
          src="test-file-stub"
        />
      </button>
      <p>
        Reserve
      </p>
    </div>
    <Tile
      className="lighter-tile"
    >
      <img
        alt="2018 Chevy Tahoe SUV"
        styleName="vehicle-image"
      />
      <div
        styleName="vehicle-details-container"
      >
        <div>
          <div
            styleName="vehicle-tile-container"
          >
            <VehicleTitle
              vehicleMake="Chevy"
              vehicleModel="Tahoe"
              vehicleStatus={1}
              yearOfMake="2018"
            />
          </div>
          <VehicleCostBreakdown
            data={
              Array [
                Object {
                  "name": "",
                  "price": undefined,
                },
              ]
            }
            removeBorderBottom={true}
          />
        </div>
      </div>
    </Tile>
    <VehicleDealer
      getUserLocation={[MockFunction]}
      location={
        Object {
          "data": Object {
            "latitude": 2344,
            "longitude": -3455,
          },
        }
      }
      vehicle={
        Object {
          "availability_term": "2018-07-11T16:50:34+00:00",
          "comments": "Trailhawk Edition Diamond Black Crystal Pearlcoat Exterior 3.6L V6 Engine 8 Speed Automatic Transmission Leather & Suede Heated Cooled Seating Contrast Stitching Leather Wrapped Steering Wheel Gunmetal Interior Accents Navigation/Backup camera Bluetooth HID Headlights Power Liftgate Power Folding Mirrors",
          "current_miles": 7475,
          "current_subscription": Object {
            "subscriber_id": "a4b5",
            "subscription_period_in_days": 31,
            "total_price": 500,
          },
          "dealer_city": "Austin",
          "dealer_id": 20,
          "dealer_latitude": 30.26715,
          "dealer_longitude": -97.74306,
          "dealer_name": "Divvy Motors",
          "dealer_state": "TX",
          "dealer_street_address": "P Sherman 42, Wallaby Way",
          "dealer_zip": "78701",
          "excess_mile_charge": 0.2,
          "is_available": true,
          "is_favorite": 1,
          "max_pickup_date": "2018-04-18T12:50:34+00:00",
          "mileage_options": Array [
            Object {
              "miles": 2000,
              "mobiliti_fee": null,
              "price": 1633,
              "rights": 0,
              "status": 1,
            },
            Object {
              "miles": 200,
              "mobiliti_fee": null,
              "price": 633,
              "rights": 0,
              "status": 1,
            },
          ],
          "miles_up_to": 40000,
          "min_pickup_date": "2018-04-11T12:50:34+00:00",
          "objImages": Array [
            "https://s3-us-west-2.amazonaws.com/subscriber-ui-assets/images/jeep_large.png",
            "https://s3-us-west-2.amazonaws.com/subscriber-ui-assets/images/jeep_large.png",
            "https://s3-us-west-2.amazonaws.com/subscriber-ui-assets/images/jeep_large.png",
          ],
          "pickup_days": Array [
            "2018-07-26 00:00:00.000",
            "2018-07-27 00:00:00.000",
            "2018-07-28 00:00:00.000",
          ],
          "secondary_attributes": Object {
            "doors": 4,
            "drive_type": "4x4",
            "exterior_color": "maroon",
            "interior_color": "beige",
            "transmission": "auto",
          },
          "vehicle_id": 4,
          "vehicle_image": "jeep_large.png",
          "vehicle_make": "Chevy",
          "vehicle_model": "Tahoe",
          "vehicle_status": 1,
          "vehicle_title": "2018 Chevy Tahoe SUV",
          "vehicle_type": "SUV",
          "vin": "9JHAIH13JIAS",
          "year_of_make": "2018",
        }
      }
    />
  </div>
  <div
    styleName="right"
  >
    <div>
      Yes
    </div>
  </div>
</div>
`;
